### DISCUSSION TITLE
`requires` in `[tox]` breaks factoring out environments

### DISCUSSION THREAD

**Discussion by bittner** [ymyzk/tox-gh-actions#187] [Q&A]
**Describe the bug**
When I add a `requires` to Tox' main section the plugin stops factoring out the Tox environments. Jobs on GHA will run all Tox environments instead of just a single one.

**To Reproduce**

Add, e.g., the following to your `tox.ini`:

```ini
[tox]
envlist = py3{5,6,7,8,9,10,11}
requires = virtualenv<20.22.0

[gh-actions]
python =
    2.7: py27
    3.5: py35
    3.6: py36
    3.7: py37
    3.8: py38
    3.9: py39
    3.10: py310
    3.11: py311
```
Irritatingly, this behavior is not consistent across all jobs spawned on GHA.

See [bittner/pyclean#66](https://github.com/bittner/pyclean/actions/runs/5717657613/job/15491833883) (Actions) for an example of failing test jobs.

**Expected behavior**
tox-gh-actions should produce separate jobs on GHA, running tests for only a single Tox environment (e.g. Python version).

**Comment by ymyzk** [comment]
Could you try adding `tox-gh-actions` to `requires` as well? See https://github.com/ymyzk/tox-gh-actions/issues/102#issuecomment-1002576825 for more context on this.

This is explained in README for tox-gh-actions v2, but not in one for v3. I'll also make sure to update the README.
